### PR TITLE
[scripts] updated get-badges analytics script

### PR DESCRIPTION
**Pull request by alexey-petrov**
closes https://github.com/sparkletown/internal-sparkle-issues/issues/1649

Data is now extracted in 3 different files

Change log:
Unique venue list is extracted into `uniqueVenues.csv`
All visited venues - `allVisits.csv`
All the necessary user data - `allData.csv`
Venue visit time format changed to `HH:MM:SS`

you can run a new script using this:
cd scripts - to make sure you’re in the scripts directroy
./get-badges venue1,venue2, … venueX path/to/KEY_FILE
where:
venue1,venue2 - venue ids you need for the analytics ( not limited to any number of venues, each separated with a comma, no space )
path/to/KEY_FILE- file with sparkle env keys, usually named some gibberish.json like sparkle-2a-firebase-adminsdk-tn7zf-0965cc48f9.json

what you’ll end up with is a reportData.csv file that looks exactly like this: https://docs.google.com/spreadsheets/d/1MTTOjPL34Sfmhzb_a0eRhaHGMcEuGijF2sexnagr6YE/edit#gid=0
One downside is that you might need to remove unnecessary venue columns from this file ( once imported to spreadsheet or w/e )
The reason being is that we’re having different ids for firebase’ user visits and enteredVenueIds
I think if we can manage a migration so that visits also includes an actual id instead of the name that would make running this script even better

### CODE REVIEW

**Comment by whoaitsaimz** [comment]
@colinhowe I am curious if this is @alexey-petrov's latest script that he has been using to provide us with some good analytics the past few months? I just ran the existing get-badges and it seemed like the older one. Perhaps as this is a script we can get this one merged through?

**Comment by colinhowe** [comment]
Re-opening this PR. It is the updated script and this needs updating+merging.

**Review by mike-lvov — CHANGES_REQUESTED** [review]

File: scripts/get-badges.ts [hunk 1]
```diff
@@ -29,21 +30,42 @@ if (!projectId || !venueIds) {
...
+              );
+
+            return { user, visits };
+          })
+        );
+    })
+    .map((el) => el.then((res) => res.flat()))
+    .flat();
+  return Promise.all(await dto);
+};
```
[the remark this answers is not in the document]
Comment (alexey-petrov) [reply scripts/get-badges.ts:67]: I'm merging duplicated users here
image

File: scripts/get-badges.ts [hunk 2]
```diff
@@ -29,21 +30,42 @@ if (!projectId || !venueIds) {
...
-  process.exit(1);
-}
-
 initFirebaseAdminApp(projectId, {
   credentialPath: credentialPath
     ? resolve(__dirname, credentialPath)
     : undefined,
 });
 
+const getUsersWithWisits = async () => {
```
[the remark this answers is not in the document]
Comment (alexey-petrov) [reply scripts/get-badges.ts]: done a065a42

**Review by colinhowe — APPROVED** [review]

**Pull request merged**